### PR TITLE
SAR PDF: Initiatives, Expenditures & Objectives

### PR DESCRIPTION
**Pull request by ailZhou**
### Description

This ticket covers the PDF generation for the initiative components in SAR. The Expenditure section needs a table to be generated so there's a new script called ExportEntityDetailsTable.tsx that handles that. Eventually we may want to consolidate some of the components for generating a table into its own util script. 

Note:
- There's some missing nested data for expenditure, will be address once the nested field is fixed.
- I know coverage is low, I will cover it in another branch
- Have at least one transition benchmark selected. @ntsummers1 found this bug caused by the R E & T tables where no transition benchmark selected will error out the pdf. I will address that in a separate branch

### Related ticket(s)

CMDCT-3107
CMDCT-3106

---
### How to test

1) Sign into MFP, any state user
2) Fill out a WP and submit
3) Sign out and login as admin user
4) Approve the previously submitted WP
5) Signout and sign back in as the state user
6) Create a SAR, and enter data for the Initiative section
7) Go to Review & Submit section, click Review PDF
8) Look to see that the PDF for Initiative section matches the mock ups: https://www.figma.com/file/oOJMUZerbafZefLQQ33GwI/MFP_Playground?type=design&node-id=2457-45591&mode=design&t=6GNcoOIopGRHuyAl-4

### Important updates

---
### Author checklist

- [ ] I have performed a self-review of my code
- [ ] I have added [thorough](https://shorturl.at/aejkF) tests, if necessary
- [ ] I have updated relevant documentation, if necessary
---

_convert to a different template: [test → val](?expand=1&template=test-to-val-deployment.md)_ | _[val → prod](?expand=1&template=val-to-prod-deployment.md)_

### CODE REVIEW

**Review by cassandradanger — COMMENTED** [review]
small styling issue, but i'm cool if it's addressed separately. It looks like the indicator text in the figma is a little bit wider than it is currently 

current:
Screenshot 2024-01-26 at 2 05 49 PM


figma:
Screenshot 2024-01-26 at 2 05 54 PM

**Review by karla-vm — CHANGES_REQUESTED** [review]
Looking great! Just a couple of styling and content changes based on Figma:

The sections are missing a subtitle under their respective headings:
Screenshot 2024-01-26 at 3 08 37 PM

Screenshot 2024-01-26 at 3 08 46 PM

Screenshot 2024-01-26 at 3 17 42 PM

Screenshot 2024-01-26 at 3 11 38 PM

And for the `Initiatives progress` table, the **Indicator** column, second row is styled differently and has a bit more content:
 
Screenshot 2024-01-26 at 3 10 16 PM

**Review by cassandradanger — COMMENTED** [review]
Looks like we're missing some hint text in the PDF (initiative progress) 
current:
Screenshot 2024-01-26 at 2 18 45 PM

Figma:
Screenshot 2024-01-26 at 2 18 35 PM

**Comment by keeysnc** [comment]
This was a lot, looks good though! Karla already mentioned some styling things I came across but thats about it. Last time I spoke with Kim she mentioned the tables needing aria labels. Not sure when we should revisit that but seems like it should be its own ticket.

**Review by keeysnc — APPROVED** [review]
lgtm!

**Comment by ntsummers1** [comment]
Screenshot 2024-01-29 at 3 55 06 PM
Screenshot 2024-01-29 at 3 55 12 PM


Noticed this when entering in some data for an expenditures

**Comment by ntsummers1** [comment]
Screenshot 2024-01-29 at 3 59 02 PM

Also got an undefined title here for objectiveCards

File: services/ui-src/src/components/cards/EntityCardBottomSection.tsx [hunk 1]
```diff
@@ -1,49 +1,153 @@
...
+
+export const EntityStepCardBottomSection = ({
+  stepType,
+  formattedEntityData,
+  printVersion,
+}: Props) => {
   const { report } = useStore() ?? {};
   switch (stepType) {
-    case "evaluationPlan":
+    case "objectiveProgress":
```
Comment (ntsummers1) [inline services/ui-src/src/components/cards/EntityCardBottomSection.tsx:67]:
```suggestion
    case OverlayModalStepTypes.OBJECTIVE_PROGRESS:
```

File: services/ui-src/src/components/cards/EntityCardBottomSection.tsx [hunk 2]
```diff
@@ -1,49 +1,153 @@
...
       if (report?.reportType === ReportType.SAR) {
```
[the remark this answers is not in the document]
Comment (ailZhou) [reply services/ui-src/src/components/cards/EntityCardBottomSection.tsx]: removed in latest commit

File: services/ui-src/src/components/cards/EntityCardBottomSection.tsx (hunk 1 above)
Comment (ailZhou) [reply services/ui-src/src/components/cards/EntityCardBottomSection.tsx]: added in latest commit

**Comment by ailZhou** [comment]
> Screenshot 2024-01-29 at 3 55 06 PM Screenshot 2024-01-29 at 3 55 12 PM
> 
> Noticed this when entering in some data for an expenditures

I'm going to extract this one out to its own ticket since it requires a bit more of a structural change than what we currently have for displaying nested children

**Review by ntsummers1 — DISMISSED** [review]

File: services/app-api/forms/sar.json [hunk 3]
```diff
@@ -960,7 +960,7 @@
...
+        "dashboardSubtitle": "Your initiatives are auto-populated from your most recent approved MFP Work Plan. Note: Initiative topics with * are required initiatives.<br><br>Recipients must report on the progress of initiatives that were ongoing during the current reporting period. For each initiative, enter information on expenditures and activities, whether continuing from prior reporting periods or initiated during this reporting period.<br><br>For each initiative, recipients must also report on progress toward achieving the objective(s) identified in the MFP Work Plan. Progress toward these objectives indicates the state or territory’s greater ability to provide home and community-based services (HCBS) instead of services in institutional settings. <br><br>If your state or territory has not achieved the targets for performance measures or expected time frames for deliverables set in the MFP Work Plan, use the following questions to explain the barriers or challenges that have hindered progress and describe plans to address them.",
```
Comment (gmrabian) [inline services/app-api/forms/sar.json:963]: I think I recall we added an aria label to asterisks in other places like for required fields. VoiceOver called it a "star".
The WP, for example, has some text like this
`<span aria-label='asterisk'>*</span>`

Just checking if that's needed here too! maybe a design question?

**Review by gmrabian — COMMENTED** [review]
one comment about accessibility on asterisks (*)

Verified visuals look good, match Figma as far as I can tell

Lot of code here that would take me a long time to get acquainted with. I'd appreciate a walkthrough but I also trust others who know this code better.

File: services/app-api/forms/sar.json (hunk 3 above)
Comment (ailZhou) [reply services/app-api/forms/sar.json]: this is a good catch and something i wasn't aware of, its added now. Thanks!

**Review by gmrabian — APPROVED** [review]
approving visuals!

**Review by cassandradanger — APPROVED** [review]
lgtm!

**Pull request merged**